### DISCUSSION TITLE
So, just in case this helps someone else with the server implimentation

### DISCUSSION THREAD

**Discussion by pjsgsy** [jchristn/WatsonTcp#157] [General]
So, totally new to this, but I just spend an hour figuring out the way to send the response BACK to the client from the server. After figuring out, of course, you need to get the connected IP/Port from the ClientConnected event and use that in conjunction with server.send.

I know, I know. Probably totally obvious to most of you, but might help someone else  :)

**Comment by jchristn** [comment]
Hi @pjsgsy the ```MessageReceivedEventArgs``` and ```StreamReceivedEventArgs``` also have the ```IpPort``` property of the sender.  So if you needed to respond to the one that sent the message, you could use that, rather than rely on retaining data from the ```ClientConnected``` event.
```csharp
static void MessageReceived(object sender, MessageReceivedEventArgs args)
{
  server.Send(args.IpPort, "Hello, sender!");
}
```
